### PR TITLE
Implement Import and Export commands for CLI Client (#300).

### LINKED ISSUE #300: Add import and export to client CLI
Import and export should be added to the client CLI.

Export should be relatively straightforward.

Import should simply generate a secret and send it to the server. We can implement custom secrets imports in the future.

### LINKED ISSUE #320: Client CLI Crashes When Printing Keys after Remote Gen Operation
The client CLI crashes with the following set of commands. Assume you are starting from a new server. Client CLI session:
```shell
| reg hackerman 1234
Logged in to hackerman
> ls
User not found
> g    
Stored: 1
> ls
name: 1
key_id: 90dee2b5b8acf5c617b7960b25c04f37cf02ef1037bafc3cb328f5b28499e5b4
data: Arbitrary Key - be93c1c792433778006c10e3359b99c47db11a0b299d8c6a163618c4ab9b5c1e


> remote-generate
Stored: 2
> ls
name: 2
Error converting LocalStorage to bytes
```

Followed by this backtrace:
```
   0:     0x557262671f2d - std::backtrace_rs::backtrace::libunwind::trace::h9135f25bc195152c
                               at /rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/std/src/../../backtrace/src/backtrace/libunwind.rs:93:5
   1:     0x557262671f2d - std::backtrace_rs::backtrace::trace_unsynchronized::h015ee85be510df51
                               at /rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/std/src/../../backtrace/src/backtrace/mod.rs:66:5
   2:     0x557262671f2d - std::sys_common::backtrace::_print_fmt::h5fad03caa9652a2c
                               at /rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/std/src/sys_common/backtrace.rs:66:5
   3:     0x557262671f2d - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h2b42ca28d244e5c7
                               at /rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/std/src/sys_common/backtrace.rs:45:22
   4:     0x55726269485c - core::fmt::write::h401e827d053130ed
                               at /rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/core/src/fmt/mod.rs:1198:17
   5:     0x55726266cc71 - std::io::Write::write_fmt::hffec93268f5cde32
                               at /rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/std/src/io/mod.rs:1672:15
   6:     0x5572626736f5 - std::sys_common::backtrace::_print::h180c4c706ee1d3fb
                               at /rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/std/src/sys_common/backtrace.rs:48:5
   7:     0x5572626736f5 - std::sys_common::backtrace::print::hd0c35d18765761c9
                               at /rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/std/src/sys_common/backtrace.rs:35:9
   8:     0x5572626736f5 - std::panicking::default_hook::{{closure}}::h1f023310983bc730
                               at /rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/std/src/panicking.rs:295:22
   9:     0x557262673411 - std::panicking::default_hook::h188fec3334afd5be
                               at /rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/std/src/panicking.rs:314:9
  10:     0x557262673c86 - std::panicking::rust_panic_with_hook::hf26e9d4f97b40096
                               at /rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/std/src/panicking.rs:698:17
  11:     0x557262673b77 - std::panicking::begin_panic_handler::{{closure}}::hfab912107608087a
                               at /rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/std/src/panicking.rs:588:13
  12:     0x557262672424 - std::sys_common::backtrace::__rust_end_short_backtrace::h434b685ce8d9965b
                               at /rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/std/src/sys_common/backtrace.rs:138:18
  13:     0x5572626738a9 - rust_begin_unwind
                               at /rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/std/src/panicking.rs:584:5
  14:     0x557261c07fc3 - core::panicking::panic_fmt::ha6dc7f2ab2479463
                               at /rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/core/src/panicking.rs:142:14
  15:     0x55726266be21 - std::io::stdio::print_to::h25dd002df6240e93
                               at /rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/std/src/io/stdio.rs:1015:9
  16:     0x55726266be21 - std::io::stdio::_print::h235f89a7e7e6b2bb
                               at /rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/std/src/io/stdio.rs:1027:5
  17:     0x557261c70b47 - lock_keeper_client_cli::storage::UserStore::list::hee7d61acc7abeaa4
                               at /home/omar/Programming/key-mgmt/lock-keeper-client-cli/src/storage.rs:192:13
  18:     0x557261d81eef - lock_keeper_client_cli::storage::Storage::list::h8f73a252ee7f23ff
                               at /home/omar/Programming/key-mgmt/lock-keeper-client-cli/src/storage.rs:94:9
  19:     0x557261c5ab53 - lock_keeper_client_cli::command::Command::execute::{{closure}}::h49e251ab34d6a046
                               at /home/omar/Programming/key-mgmt/lock-keeper-client-cli/src/command.rs:212:17
  20:     0x557261db0729 - <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll::h390548121572b262
                               at /rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/core/src/future/mod.rs:91:19
  21:     0x557261d801dc - lock_keeper_client_cli::app::run::{{closure}}::h34566f5daa02ae5f
                               at /home/omar/Programming/key-mgmt/lock-keeper-client-cli/src/app.rs:23:52
  22:     0x557261dafc69 - <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll::h1a1faa37774674ba
                               at /rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/core/src/future/mod.rs:91:19
  23:     0x557261d86535 - lock_keeper_client_cli::main::{{closure}}::hd49627a30ba91aca
                               at /home/omar/Programming/key-mgmt/lock-keeper-client-cli/src/main.rs:15:39
  24:     0x557261db1b5b - <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll::h76800b1a25d39226
                               at /rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/core/src/future/mod.rs:91:19
  25:     0x557261ce37e0 - tokio::park::thread::CachedParkThread::block_on::{{closure}}::h03d7d203e1d7bfbd
                               at /home/omar/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.21.2/src/park/thread.rs:267:54
  26:     0x557261d703a0 - tokio::coop::with_budget::{{closure}}::ha24267122963054d
                               at /home/omar/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.21.2/src/coop.rs:102:9
  27:     0x557261d08bf1 - std::thread::local::LocalKey<T>::try_with::h49994f2063d91480
                               at /rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/std/src/thread/local.rs:445:16
  28:     0x557261d0894d - std::thread::local::LocalKey<T>::with::h2dd7734adb3f1512
                               at /rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/std/src/thread/local.rs:421:9
  29:     0x557261ce3676 - tokio::coop::with_budget::h83d753ceec89d37f
                               at /home/omar/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.21.2/src/coop.rs:95:5
  30:     0x557261ce3676 - tokio::coop::budget::ha30c942d91b04357
                               at /home/omar/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.21.2/src/coop.rs:72:5
  31:     0x557261ce3676 - tokio::park::thread::CachedParkThread::block_on::h5e86dd4318397580
                               at /home/omar/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.21.2/src/park/thread.rs:267:31
  32:     0x557261d9c3a9 - tokio::runtime::enter::Enter::block_on::hd935eb72dc56e39c
                               at /home/omar/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.21.2/src/runtime/enter.rs:152:13
  33:     0x557261cfdf3c - tokio::runtime::scheduler::multi_thread::MultiThread::block_on::hfd7839d0504389cd
                               at /home/omar/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.21.2/src/runtime/scheduler/multi_thread/mod.rs:79:9
  34:     0x557261d9f670 - tokio::runtime::Runtime::block_on::hdcc015d2784796a2
                               at /home/omar/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.21.2/src/runtime/mod.rs:492:44
  35:     0x557261d44f55 - lock_keeper_client_cli::main::h6f0fcbe7f5d8d03b
                               at /home/omar/Programming/key-mgmt/lock-keeper-client-cli/src/main.rs:15:5
  36:     0x557261c442eb - core::ops::function::FnOnce::call_once::h01df62ed2c522c38
                               at /rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/core/src/ops/function.rs:248:5
  37:     0x557261c7df7e - std::sys_common::backtrace::__rust_begin_short_backtrace::ha1e492abda9a3eb9
                               at /rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/std/src/sys_common/backtrace.rs:122:18
  38:     0x557261c96631 - std::rt::lang_start::{{closure}}::h57899d5dbf430576
                               at /rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/std/src/rt.rs:145:18
  39:     0x5572626671fe - core::ops::function::impls::<impl core::ops::function::FnOnce<A> for &F>::call_once::hcdfee62722e5e4b8
                               at /rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/core/src/ops/function.rs:280:13
  40:     0x5572626671fe - std::panicking::try::do_call::h84ca51609826746f
                               at /rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/std/src/panicking.rs:492:40
  41:     0x5572626671fe - std::panicking::try::hd58075e533b8e0cb
                               at /rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/std/src/panicking.rs:456:19
  42:     0x5572626671fe - std::panic::catch_unwind::h1ebac24d83cb6ce2
                               at /rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/std/src/panic.rs:137:14
  43:     0x5572626671fe - std::rt::lang_start_internal::{{closure}}::h0145388a1edd1640
                               at /rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/std/src/rt.rs:128:48
  44:     0x5572626671fe - std::panicking::try::do_call::h7630182e534a0a32
                               at /rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/std/src/panicking.rs:492:40
  45:     0x5572626671fe - std::panicking::try::h05b6544f0c6331dc
                               at /rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/std/src/panicking.rs:456:19
  46:     0x5572626671fe - std::panic::catch_unwind::h77b2ba8fd3309f34
                               at /rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/std/src/panic.rs:137:14
  47:     0x5572626671fe - std::rt::lang_start_internal::h6612c8a7a6861b8b
                               at /rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/std/src/rt.rs:128:20
  48:     0x557261c96600 - std::rt::lang_start::hac3f9c2aa793335c
                               at /rustc/a55dd71d5fb0ec5a6a3a9e8c27b2127ba491ce52/library/std/src/rt.rs:144:17
  49:     0x557261d44ffc - main
  50:     0x7fb62532ed90 - __libc_start_call_main
                               at ./csu/../sysdeps/nptl/libc_start_call_main.h:58:16
  51:     0x7fb62532ee40 - __libc_start_main_impl
                               at ./csu/../csu/libc-start.c:392:3
  52:     0x557261c08315 - _start
  53:                0x0 - <unknown>
```

### PR DESCRIPTION
**Pull request by gatoWololo**
Implement Import and Export commands for CLI Client. Closes #300. Closes #320 

- Updates client API docstrings.

### CODE REVIEW

File: lock-keeper-client-cli/src/command.rs [hunk 1]
```diff
@@ -228,8 +311,8 @@ impl FromStr for Command {
     fn from_str(s: &str) -> Result<Self, Self::Err> {
         let mut split = s.trim().split(' ');
 
-        match split.next() {
-            Some(command) => match command {
+        let command: Command = match split.next() {
```
Comment (picklenerd) [inline lock-keeper-client-cli/src/command.rs:314]: Does this type need to be specified?

**Comment by picklenerd** [comment]
This looks great! Thanks for updating the docs and cleaning up some of the messiness in the CLI crate!

File: lock-keeper/src/crypto/arbitrary_secret.rs [hunk 2]
```diff
@@ -96,6 +96,18 @@ impl Secret {
...
     }
+
+    /// Retrieve key material.
+    ///
+    /// Security: It may seem like a security concern to have a method to access
+    /// the key material. However:
+    /// 1) This type is Serialize and Deserialize. So users can already extract
+    /// this material by serializing the data.
+    /// 2) Nothing stops us from ignoring memory safety and reading the bytes
+    /// directly. This method gives no more access.
```
[the remark this answers is not in the document]
Comment (marsella) [reply lock-keeper/src/crypto/arbitrary_secret.rs:107]: Context clarification: is this only used in the client API to print out the secret? If so, can you implement `Display` for this type instead? Or maybe put it behind a feature flag that indicates it's for use in demos and sample code but not as a method called in protocol implementations?

The goal of keeping all the secret types so locked down / with limited APIs is to build in misuse resistance. You all are careful developers, but a powerful method with a generic name is easy to misuse, e.g. in a case where maybe the spec isn't super clear or a developer is working on a deadline and trying to get something to work fast.

File: lock-keeper-client-cli/src/command.rs (hunk 1 above)
Comment (gatoWololo) [reply lock-keeper-client-cli/src/command.rs:314]: No, it doesn't. Sometimes I like specifying the time for clarity. But I noticed most of the codebase prefers not listing the types. I'm okay with whichever people prefer.

File: lock-keeper/src/crypto/arbitrary_secret.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (gatoWololo) [reply lock-keeper/src/crypto/arbitrary_secret.rs]: I have updated the code. The public facing `Secret` type now implements `Display` the private inner type still has access to the bytes, but this type can only be used by us from within the crate.

**Review by picklenerd — APPROVED** [review]

**Pull request merged**